src/App.js

```diff
@@ -474,7 +484,22 @@ class App extends React.Component {
 }
 
 export default function(props) {
-  const theme = useTheme()
   const { account } = useWallet()
-  return <App theme={theme} walletAccount={account} {...props} />
+  const theme = useTheme()
+  const themeMode = useThemeMode()
+  const clientTheme = useClientTheme()
+  const { appearance } = clientTheme
+
+  useEffect(() => {
+    themeMode.set(appearance)
+  }, [appearance, themeMode])
+
+  return (
+    <App
+      clientTheme={clientTheme}
```

Comment:
I find having both `clientTheme` and `theme` next to each other quite confusing.

Perhaps we could find another name for `clientTheme`, or perhaps better, we could isolate the transition-related UI to its own component?